File: basic-crud/src/main/java/com/sowfte/basiccrud/service/UserService.java
package com.sowfte.basiccrud.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import com.sowfte.basiccrud.entity.UserEntity;
import com.sowfte.basiccrud.repository.UserRepository;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository ;

    // create user
    public UserEntity createUser(UserEntity userRecord){
       return userRepository.save(userRecord);
    }

    // get all user
    public List<UserEntity> getAllUser(){
        return userRepository.findAll();
    }

    // get user by id
    public UserEntity getUserByID(Long id){
        return userRepository.findById(id).get();
    }

    // delete user by id
    public String deleteUserByID(Long id){
        if(!userRepository.existsById(id)){
            return "User does not exist";
        }
        userRepository.deleteById(id);
        return "User successfully deleted";
    }

    // update a user by id
    public UserEntity updateUserByID(UserEntity newUserRecord, Long id){
        newUserRecord.setId(id);
         UserEntity updatedRecord = userRepository.save(newUserRecord);

         return ResponseEntity.ok(updatedRecord).getBody();
    }
    
}

File: Nodejs Basic Server/services/userService.js
const User = require("../model/User");

// create user
async function createUser(payload) {
  return await User.create({ ...payload });
}

// get all user

async function getAllUsers() {
  return await User.findAll();
}

// get user by id
async function getUserByID(id) {
  return await User.findByPk(id);
}

// delete user by id
async function deleteUserByID(id) {
  const user = await this.getUserByID(id);
  if (!user) {
    return "User not found";
  }
  await user.destroy();
  return "User deleted !";
}

// update user by id
async function updateUser(id, payload) {
  const user = await this.getUserByID(id);
  if (!user) {
    return "User is not found";
  }
  return user.update({ ...payload });
}

module.exports = {
  createUser,
  getAllUsers,
  getUserByID,
  deleteUserByID,
  updateUser,
};
